src/ripple/overlay/impl/PeerSet.cpp

```diff
@@ -110,53 +123,26 @@ PeerSet::invokeOnTimer()
         setTimer();
 }
 
-bool
-PeerSet::isActive()
-{
-    ScopedLockType sl(mLock);
-    return !isDone();
-}
-
 void
 PeerSet::sendRequest(
     const protocol::TMGetLedger& tmGL,
     std::shared_ptr<Peer> const& peer)
 {
-    if (!peer)
-        sendRequest(tmGL);
-    else
-        peer->send(std::make_shared<Message>(tmGL, protocol::mtGET_LEDGER));
-}
-
-void
-PeerSet::sendRequest(const protocol::TMGetLedger& tmGL)
-{
-    ScopedLockType sl(mLock);
-
-    if (mPeers.empty())
-        return;
-
     auto packet = std::make_shared<Message>(tmGL, protocol::mtGET_LEDGER);
 
-    for (auto id : mPeers)
+    if (peer)
```

Comment:
No change needed (and not much you can do at this point, since other code locks you into this API) but just noting that I think that it's best to be explicit about things:

```
/** Send request to all peers */
void
PeerSet::broadcastRequest(
    const protocol::TMGetLedger& tmGL);

/** Send request to the specified peer */
void
PeerSet::sendRequest(
    const protocol::TMGetLedger& tmGL,
    std::shared_ptr<Peer> const& peer);
```